### PR TITLE
set base-href dynamically

### PR DESCRIPTION
**Pull request by rayene**

### CODE REVIEW

File: api/routing/routes.go [hunk 1]
```diff
@@ -68,7 +37,7 @@ func NewRouter(tm db.TransactionManager, version, hateoasBaseBath, uiBasePath st
 	tonic.SetErrorHook(hateoas.ErrorHook(jujerr.ErrHook))
 
 	// redirect root routes to angular assets
-	router.Use(gzip.Gzip(gzip.DefaultCompression), static.Serve("/ui", static.LocalFile("./webui", true)))
+	//router.Use(gzip.Gzip(gzip.DefaultCompression), static.Serve("/ui", static.LocalFile("./webui", true)))
```
Comment (FabienM) [inline api/routing/routes.go]: Dead code + it disables gzip compression (which is bad).

File: api/routing/routes.go [hunk 2]
```diff
@@ -111,8 +80,17 @@ func NewRouter(tm db.TransactionManager, version, hateoasBaseBath, uiBasePath st
 	unsecured.GET("/openapi.json", nil, router.OpenAPI(infos, "json"))
 	unsecured.GET("/openapi.yaml", nil, router.OpenAPI(infos, "yaml"))
 
-	// redirect unknown routes to angular
-	router.Engine().NoRoute(uiRedirectHandler(uiBasePath))
+	if _, err := os.Stat(webUIDir + "/index.html"); os.IsNotExist(err) {
+		log.Warn("index.html not found. Starting in API only mode. Nop static content will be served.")
```
Comment (FabienM) [inline api/routing/routes.go]: `s/Nop/No`

Comment (FabienM) [reply api/routing/routes.go]: add `return router` here and remove the `else {` to keep a single code path

Comment (rayene) [reply api/routing/routes.go]: Done

File: docker/main/Dockerfile [hunk 3]
```diff
@@ -4,14 +4,16 @@ COPY dist /app
...
+                /app/appcatalog \
+    && apt-get update \
+    && apt-get install -y curl wget ca-certificates
+
 USER nobody
 
 EXPOSE 8081
 
-CMD ./appcatalog-configuration --debug --output=./config.json start && ./appcatalog.sh --config=./config.json
+CMD ./appcatalog-configuration --debug --output=./config.json start && ./appcatalog --config=./config.json
```
Comment (yroffin) [inline docker/main/Dockerfile]: why did you remove .sh ... it's for loading mycompany.sh if needed for one ENV ?

File: webui/src/app/app.module.ts [hunk 4]
```diff
@@ -258,12 +260,6 @@ export function createTranslateLoader(http: HttpClient) {
     })
   ],
   providers: [
-    /**
-     * Base Href
-     */
-    {
-      provide: APP_BASE_HREF, useValue: environment.baseHref
```
Comment (yroffin) [inline webui/src/app/app.module.ts]: i'm suprised of this code removing, must explain it tomorow

File: docker/main/Dockerfile (hunk 3 above)
Comment (rayene) [reply docker/main/Dockerfile]: It contains a bug. The "/api" route is appearing twice in the appcatalog.sh file. I preferred deactivating it temporarily.

**Comment by yroffin** [comment]
Good job
